tracer/src/Datadog.Trace/IAST/Evidence.cs

```diff
@@ -5,22 +5,52 @@
 
 #nullable enable
 
+using System.Collections.Generic;
+
 namespace Datadog.Trace.Iast;
 
 internal readonly struct Evidence
 {
+    private readonly Range[]? _ranges;
+
     public Evidence(string value, Range[]? ranges = null)
     {
         this.Value = value;
-        this.Ranges = ranges;
+        this._ranges = ranges;
     }
 
     public string Value { get; }
 
-    public Range[]? Ranges { get; }
+    // This method is only used once when serializing to json (and it is also called from unit tests)
+    public List<ValuePart>? ValueParts => GetValuePartsFromRanges();
+
+    private List<ValuePart>? GetValuePartsFromRanges()
```

Comment:
nit: not a big fan of this pattern, as it's not entirely obvious you're doing a bunch of work when calling this. e.g. in your test your doing

```csharp
var ev = new Evidence("sql_query", ranges);
ev.ValueParts.Count.Should().Be(2);
ev.ValueParts[0].Source.Should().Be(0);
ev.ValueParts[1].Source.Should().Be(1);
ev.ValueParts[0].Value.Should().Be("sq");
ev.ValueParts[1].Value.Should().Be("l_");
```

Which calls `GetValuePartsFromRanges()` 5 times. If you _only_ had a `GetValueParts()` method, you probably wouldn't have done that in the unit test, as it would have been obvious 😉

Reply:
This pattern was used for simple automatic serialization. By including GetValueParts as a property, it is added to the json. It is the only call that should be done to this property (besides unit tests).

Reply:
Ah ok, I see, makes sense. In which case maybe the tests should just be doing

```csharp
var valueParts = ev.ValueParts;
```

To keep my OCD happy 😅